In [ ]:
#if using html
#import ssl
#ssl._create_default_https_context = ssl._create_unverified_context

In [ ]:
#Importing the libraries
import numpy as np
import pandas as pd

In [ ]:
#Data preparation
#data source -->

Path = "/content/CRIME_OCCURRED.csv"

In [ ]:
Path

'/content/CRIME_OCCURRED.csv'

In [ ]:
#from google.colab import drive
#rive.mount('/content/drive')

In [ ]:
#Storing all data in data variable
data = pd.read_csv(Path)
data.head()

Area_Name  Year           Group_Name Sub_Group_Name  \
0  Lakshadweep  2111  Burglary - Property    3. Burglary   
1      Manipur  2112  Burglary - Property    3. Burglary   
2  Lakshadweep  2113  Burglary - Property    3. Burglary   
3      Manipur  2113  Burglary - Property    3. Burglary   
4  Lakshadweep  2112  Burglary - Property    3. Burglary   

   Cases_Property_Recovered  Cases_Property_Stolen  \
0                         1                      1   
1                         1                     44   
2                         1                      6   
3                         1                     68   
4                         2                      9   

   Value_of_Property_Recovered  Value_of_Property_Stolen  
0                            1                     12111  
1                         9525                   4136635  
2                            1                     65111  
3                         1811                   3149166  
4                        26795                    115645

In [ ]:
#a brief about all the columns in dataset
data.columns

Index(['Area_Name', 'Year', 'Group_Name', 'Sub_Group_Name',
       'Cases_Property_Recovered', 'Cases_Property_Stolen',
       'Value_of_Property_Recovered', 'Value_of_Property_Stolen'],
      dtype='object')

Now as some data columns are strings(words) so first we have to convert them into integers and we will do it by encoding techniques by using a label encoder

In [ ]:
data.head()

Area_Name  Year           Group_Name Sub_Group_Name  \
0  Lakshadweep  2111  Burglary - Property    3. Burglary   
1      Manipur  2112  Burglary - Property    3. Burglary   
2  Lakshadweep  2113  Burglary - Property    3. Burglary   
3      Manipur  2113  Burglary - Property    3. Burglary   
4  Lakshadweep  2112  Burglary - Property    3. Burglary   

   Cases_Property_Recovered  Cases_Property_Stolen  \
0                         1                      1   
1                         1                     44   
2                         1                      6   
3                         1                     68   
4                         2                      9   

   Value_of_Property_Recovered  Value_of_Property_Stolen  
0                            1                     12111  
1                         9525                   4136635  
2                            1                     65111  
3                         1811                   3149166  
4                        26795                    115645

In [ ]:
#converting string to integer
from sklearn import preprocessing
label = preprocessing.LabelEncoder()
data['Area_Name'] = label.fit_transform(data['Area_Name'])
data['Cases_Property_Recovered'] = label.fit_transform(data['Cases_Property_Recovered'])
data['Cases_Property_Stolen'] = label.fit_transform(data['Cases_Property_Stolen'])
data['Year'] = label.fit_transform(data['Year'])
data.head()

Area_Name  Year           Group_Name Sub_Group_Name  \
0         18     1  Burglary - Property    3. Burglary   
1         21     2  Burglary - Property    3. Burglary   
2         18     3  Burglary - Property    3. Burglary   
3         21     3  Burglary - Property    3. Burglary   
4         18     2  Burglary - Property    3. Burglary   

   Cases_Property_Recovered  Cases_Property_Stolen  \
0                         0                      0   
1                         0                     11   
2                         0                      2   
3                         0                     23   
4                         1                      3   

   Value_of_Property_Recovered  Value_of_Property_Stolen  
0                            1                     12111  
1                         9525                   4136635  
2                            1                     65111  
3                         1811                   3149166  
4                        26795                    115645

Defining X and Y
And now I have created a list in which I think it will be the decidingfactor for a *Crime Prediction* i.e. *Area_Name, Year, Cases_Property_Recovered* and assign it to variable features now I will pass features in the dataset and store it as x and the *Value_of_Property_Stolen* of the car as y.

In [ ]:
#deciding feactor, features
features = ["Area_Name","Year", "Cases_Property_Recovered","Cases_Property_Stolen"]
x = data[features]
y = data["Value_of_Property_Stolen"]

I believe that these x parameters are more appropriate, and if you want to modify the parameters because you believe they are relevant, you can do so.


###Making the model
Splitting the Dataset
We must first import the test train split from sklearn model selection before splitting our model dataset into a train and test dataset.




In [ ]:
from sklearn.model_selection import train_test_split

In [ ]:
train_x,test_x,train_y,test_y = train_test_split(x,y)

###Model
here we will be making a decision tree regressor model because we want our value to be an integer value and naming our model tree.

In [ ]:
from sklearn.tree import DecisionTreeRegressor

In [ ]:
tree = DecisionTreeRegressor()
tree.fit(train_x,train_y)

DecisionTreeRegressor()

###Predicting
nowhere I have made a variable named prediction which will predict the values of our test data.

In [ ]:
prediction = tree.predict(test_x)

In [ ]:
data.head()

Area_Name  Year           Group_Name Sub_Group_Name  \
0         18     1  Burglary - Property    3. Burglary   
1         21     2  Burglary - Property    3. Burglary   
2         18     3  Burglary - Property    3. Burglary   
3         21     3  Burglary - Property    3. Burglary   
4         18     2  Burglary - Property    3. Burglary   

   Cases_Property_Recovered  Cases_Property_Stolen  \
0                         0                      0   
1                         0                     11   
2                         0                      2   
3                         0                     23   
4                         1                      3   

   Value_of_Property_Recovered  Value_of_Property_Stolen  
0                            1                     12111  
1                         9525                   4136635  
2                            1                     65111  
3                         1811                   3149166  
4                        26795                    115645

In [ ]:
data[["Area_Name","Year", "Cases_Property_Recovered","Cases_Property_Stolen"]].head()

Area_Name  Year  Cases_Property_Recovered  Cases_Property_Stolen
0         18     1                         0                      0
1         21     2                         0                     11
2         18     3                         0                      2
3         21     3                         0                     23
4         18     2                         1                      3

we also can predict entered values simply by adding the values to the list manually.

In [ ]:
tree.predict(([[18,1,0,0]]))

/usr/local/lib/python3.10/dist-packages/sklearn/base.py:439: UserWarning: X does not have valid feature names, but DecisionTreeRegressor was fitted with feature names
  warnings.warn(


array([12111.])

In [ ]:
#lets check the value
data.Value_of_Property_Stolen.head()

0      12111
1    4136635
2      65111
3    3149166
4     115645
Name: Value_of_Property_Stolen, dtype: int64

which is the first value of our dataset and the real value is 12111 and predicted value is 12111 which is 100% accurate

and also, we can predict values from our dataset i.e. here I am predicting first 5 values from dataset.

In [ ]:
tree.predict(x[:5])

array([  12111., 2184845.,   65111., 2761756.,  115645.])

##Accuracy
Now we’ll look at our model’s accuracy. Our model is 100 percent accurate, which means it correctly predicted 100 values out of a possible 100. Which is excellent.

In [ ]:
tree.score(test_x,prediction)

1.0